internal/impl/kube.go

```diff
@@ -58,15 +57,33 @@ const (
 // Start value for ports used by the public and private listeners.
 var externalPort int32 = 20000
 
-// replicaSet contains information about a replica set.
-type replicaSet struct {
-	name            string                        // name of the replica set
-	components      []*ReplicaSetConfig_Component // components hosted by replica set
-	image           string                        // name of the image to be deployed
-	namespace       string                        // namespace of the replica set
-	depId           string                        // app deployment identifier
-	app             *protos.AppConfig             // app configuration
-	traceServiceURL string                        // trace exporter URL
+// deployment contains information about a deployment of a Service Weaver
+// application.
+//
+// Note that this is different from a Kubernetes Deployment. A deployed Service
+// Weaver application consists of many Kubernetes Deployments.
+type deployment struct {
+	deploymentId    string            // globally unique deployment id
+	image           string            // Docker image URI
+	traceServiceURL string            // where traces are exported to, if not empty
+	config          *kubeConfig       // [kube] config from weaver.toml
+	app             *protos.AppConfig // parsed weaver.toml
+	nodes           []node            // nodes
+}
+
+// node contains information about a possibly replicated group of components.
+type node struct {
```

Comment:
nit: I find the `node` to be slightly confusing. When I think of a node I usually think of a physical machine, server. I would slightly prefer `group` but someone can argue that it's confusing as well.